locator.go

```diff
@@ -0,0 +1,468 @@
+			case string:
+				selector += fmt.Sprintf(" >> :scope:has-text('%s')", hasText)
+			}
+		}
+		if option.Has != nil {
+			has := option.Has.(*locatorImpl)
+			if frame != has.frame {
+				return nil, errors.New("inner 'has' locator must belong to the same frame")
+			}
+			selector += " >> has=" + has.selector
+		}
+	}
+
+	return &locatorImpl{frame: frame, selector: selector, options: option}, nil
+}
+
+func convertRegexp(reg *regexp.Regexp) string {
+	matches := regexp.MustCompile(`\(\?([imsU]+)\)(.+)`).FindStringSubmatch(reg.String())
+
+	var pattern, flags string
+	if len(matches) == 3 {
+		pattern = matches[2]
+		flags = matches[1]
+	} else {
+		pattern = reg.String()
+	}
+	return fmt.Sprintf("'%s', '%s'", pattern, flags)
+}
+
+func (l *locatorImpl) AllInnerTexts() ([]string, error) {
+	innerTexts, err := l.frame.EvalOnSelectorAll(l.selector, "ee => ee.map(e => e.innerText)")
+	if err != nil {
+		return nil, err
+	}
+	texts := innerTexts.([]interface{})
+	result := make([]string, len(texts))
+	for i := range texts {
+		result[i] = texts[i].(string)
+	}
+	return result, nil
+}
+
+func (l *locatorImpl) AllTextContents() ([]string, error) {
+	textContents, err := l.frame.EvalOnSelectorAll(l.selector, "ee => ee.map(e => e.textContent || '')")
+	if err != nil {
+		return nil, err
+	}
+	texts := textContents.([]interface{})
+	result := make([]string, len(texts))
+	for i := range texts {
+		result[i] = texts[i].(string)
+	}
+	return result, nil
+}
+
+func (l *locatorImpl) BoundingBox(options ...LocatorBoundingBoxOptions) (*LocatorBoundingBoxResult, error) {
+	var option PageWaitForSelectorOptions
+	if len(options) == 1 {
+		option.Timeout = options[0].Timeout
+	}
+
+	var result *LocatorBoundingBoxResult
+	err := l.withElement(func(handle ElementHandle) error {
+		rect, err := handle.BoundingBox()
+		if err != nil {
+			return err
+		}
+		result = &LocatorBoundingBoxResult{
+			X:      float64(rect.X),
+			Y:      float64(rect.Y),
+			Width:  float64(rect.Width),
+			Height: float64(rect.Height),
+		}
+		return nil
+	}, option)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (l *locatorImpl) Check(options ...FrameCheckOptions) error {
+	return l.frame.Check(l.selector, options...)
+}
+
+func (l *locatorImpl) Click(options ...PageClickOptions) error {
+	return l.frame.Click(l.selector, options...)
+}
+
+func (l *locatorImpl) Count() (int, error) {
+	return l.frame.queryCount(l.selector)
+}
+
+func (l *locatorImpl) Dblclick(options ...FrameDblclickOptions) error {
+	return l.frame.Dblclick(l.selector, options...)
+}
+
+func (l *locatorImpl) DispatchEvent(typ string, eventInit interface{}, options ...PageDispatchEventOptions) error {
+	return l.frame.DispatchEvent(l.selector, typ, eventInit, options...)
+}
+
+func (l *locatorImpl) DragTo(target Locator, options ...FrameDragAndDropOptions) error {
+	if len(options) == 1 {
+		strict := true
+		options[0].Strict = &strict
```

Comment:
```suggestion
		options[0].Strict = Bool(true)
```
I think we have a wrapper function for that.

Reply:
I didn't know that useful wrapper function!
fixed codes including similar to this!